src/hotspot/share/ci/ciKlass.cpp

```diff
@@ -222,7 +222,7 @@ jint ciKlass::modifier_flags() {
 jint ciKlass::access_flags() {
   assert(is_loaded(), "not loaded");
   GUARDED_VM_ENTRY(
-    return get_Klass()->access_flags().as_int();
+    return get_Klass()->access_flags().as_unsigned_short();
```

Comment:
Again it is unclear to me whether this API should also now return u2.

Reply:
I don't think it should.  I think the boundary of where we promote the u2 to int should be at this API.  If those working on the compiler code would like to propagate the size of the storage (u2) around, they can decide to do that.